clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -747,7 +747,7 @@ const llvm::fltSemantics &FP80Type::getFloatSemantics() const {
 llvm::TypeSize
 FP80Type::getTypeSizeInBits(const mlir::DataLayout &dataLayout,
                             mlir::DataLayoutEntryListRef params) const {
-  return llvm::TypeSize::getFixed(16);
+  return llvm::TypeSize::getFixed(128);
```

Comment:
I think this just caused an issue recently? cc @Lancern @PikachuHyA

Reply:
Yes, but the issue is not rooted here, it's more related to the LLVM lowering part. `FP80Type::getTypeSizeInBits` should return 128.

Reply:
Yes, I agree FP80Type::getTypeSizeInBits should return 128.

Reply:
Yes, the issue is related to the lowering of `SignBitOp`. We have temporarily skipped the handling of `FP80Type` and will address it later. https://github.com/llvm/clangir/pull/1033#issuecomment-2465484388

As for `FP80Type::getTypeSizeInBits`, I agree it should return 128 now.